### PR TITLE
Postgresql patches

### PR DESCRIPTION
**Pull request by zdennis**
# What does this PR do?

This PR back ports two patches for PostgreSQL from a downstream app allowing us to remove this patch and updates how the patches in this code-base are expected to be loaded by the rails2 repository.

# Notable things

PostgreSQL patches:

* Use `public` namespace when dumping schema information. 
* Update the `schema_search_path` to strip out white-space. This will let us remove https://github.com/Genius/rails2/pull/6/commits/649a1ca2e9dcf0d0a6b45fe36d7c0786324b8a49

Changes to loading of the patches:

* Group all patches into directories named by their Rails component (e.g. `active_record`, `active_support`, etc). This allows each component inside of rails2 to load their corresponding patches when loaded. 
* Provide shim files (e.g. `action_view`, etc) that make it easier to add patches for additional Rails components without having to modify rails2 again.

### CODE REVIEW

File: lib/rails2_ruby2.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 require "rails2_ruby2/version"
 require "rails2_ruby2/kernel_helpers"
-require "rails2_ruby2/rails_init"
```
Comment (zdennis) [inline lib/rails2_ruby2.rb:3]: This is no longer used so we nix'd it.

File: lib/rails2_ruby2/action_pack.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Dir[File.dirname(__FILE__) + '/action_pack/**/*.rb'].each do |file|
+  require file.sub(/#{Regexp.escape(File.extname(file))}$/, '')
+end
```
Comment (zdennis) [inline lib/rails2_ruby2/action_pack.rb]: Make it easier for future extensions to be loaded.

File: lib/rails2_ruby2/action_view.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Add extensions to action_view/ directory
+
+# Shim for action_view extensions
```
Comment (zdennis) [inline lib/rails2_ruby2/action_view.rb:3]: Example of a shim for components that don't have extensions today, but may in the future.

File: lib/rails2_ruby2/active_record/postgresql_adapter.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          minvalue = select_value("SELECT min_value FROM #{quoted_sequence}")
+        end
+      end
+
+      select_value <<-end_sql, 'SCHEMA'
+      SELECT setval(#{quote(quoted_sequence)}, #{max_pk ? max_pk : minvalue}, #{max_pk ? true : false})
+      end_sql
+    end
+  end
+end
```
Comment (zdennis) [inline lib/rails2_ruby2/active_record/postgresql_adapter.rb:37]: The above patches have been removed from the downstream app as well as the rails2 fork.

File: lib/rails2_ruby2/active_record/schema_statements.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require 'active_record/connection_adapters/abstract/schema_statements'
+
+ActiveRecord::ConnectionAdapters::SchemaStatements.module_eval do
+  def dump_schema_information #:nodoc:
+    sm_table = "public.#{ActiveRecord::Migrator.schema_migrations_table_name}"
+    migrated = select_values("SELECT version FROM #{sm_table} ORDER BY version")
+    migrated.map { |v| "INSERT INTO #{sm_table} (version) VALUES ('#{v}');" }.join("\n\n")
+  end
+end
```
Comment (zdennis) [inline lib/rails2_ruby2/active_record/schema_statements.rb:9]: The above patch have been removed from the downstream app.

File: lib/rails2_ruby2/action_pack.rb (hunk 2 above)
Comment (nightpool) [inline lib/rails2_ruby2/action_pack.rb]: newlines eol

File: lib/rails2_ruby2/action_pack.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Dir[File.dirname(__FILE__) + '/action_pack/**/*.rb'].each do |file|
+  require file.sub(/#{Regexp.escape(File.extname(file))}$/, '')
```
Comment (nightpool) [inline lib/rails2_ruby2/action_pack.rb]: given that we're globbing for `.rb`, i think it's safe to do `file.chomp('.rb')`

File: lib/rails2_ruby2/active_support.rb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+Dir[File.dirname(__FILE__) + '/active_support/**/*.rb'].each do |file|
+  require file.sub(/#{Regexp.escape(File.extname(file))}$/, '')
```
Comment (nightpool) [inline lib/rails2_ruby2/active_support.rb]: ibid

File: lib/rails2_ruby2/action_pack.rb (hunk 6 above)
Comment (zdennis) [reply lib/rails2_ruby2/action_pack.rb]: Good call, much simpler. Updated in 7e66adaec09d99c3942d0f8006bd158c0e80a2f6

File: lib/rails2_ruby2/action_pack.rb (hunk 2 above)
Comment (zdennis) [reply lib/rails2_ruby2/action_pack.rb]: 👍 ending newlines added to files that were missing it in 54c483e22a9c7d1eb26dd835ea44d770a1e1f8f5

**Comment by nightpool** [comment]
Corresponding pull request for Genius/rails2: https://github.com/Genius/rails2/pull/6

**Review by nightpool — CHANGES_REQUESTED** [review]
Let's add a quick note about the changes to `require` statements to the README, and bump the gem major version.

**Review by nightpool — APPROVED** [review]
approved % readme changes

**Pull request merged**